LearningMissionSimulation/PlaygroundGarush.cs

```diff
@@ -82,11 +82,11 @@ List<Module> GetModuleList()
             int maxModuleCount = 5;
             minModuleCount = Math.Min(allModuleCount, minModuleCount);
             maxModuleCount = Math.Min(allModuleCount, maxModuleCount);
-            int Count = AttributeGenerator.random.Next(minModuleCount, maxModuleCount);
+            int moduleCount = AttributeGenerator.random.Next(minModuleCount, maxModuleCount + 1);
 
-            for (int i = 0; i < Count; i++)
+            for (int i = 0; i < moduleCount; i++)
             {
-                Guid id = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count - 1)];
+                Guid id = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
```

Comment:
Your logic allows to add the same module multiple times. You should avoid that.

Reply:
fixed